### PR TITLE
[TA-3403][TA-3404]: Add X-Addresses codec support

### PR DESCRIPTION
**Pull request by GuillemGarciaDev**
# [TA-3403][TA-3404]: Add X-Addresses codec support

This PR aims to add encoding/decoding support for X-Addresses in `address-codec` package.  It includes encoding/decoding, validation and conversion functions.

## Changes

- Added `IsValidXAddress`
- Added `EncodeXAddress`
- Added `DecodeXAddress`
- Added `XAddressToClassicAddress` and `ClassicAddressToXAddress`
- Includes `EncodeAccountIDToClassicAddress` missing function on `codec.go`
- Error refactoring for `address-codec` package.
- Refactored `xrpl` `Wallet` function algorithm argument

### CODE REVIEW

**Review by florent-uzio — DISMISSED** [review]
Looks awersome! A great addition to the SDK!

File: address-codec/codec.go [hunk 1]
```diff
@@ -27,6 +25,16 @@ const (
 	FamilySeedPrefix = 0x21
 	// Node/validation public key prefix - value is 28
 	NodePublicKeyPrefix = 0x1C
+
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply address-codec/codec.go:28]: Absolutely not! Good catch

**Review by florent-uzio — DISMISSED** [review]
LGTM 👍🏻

**Review by florent-uzio — DISMISSED** [review]

File: address-codec/sha256.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"golang.org/x/crypto/ripemd160"
+)
+
+// Returns byte slice of a double hashed given byte slice.
+// The given byte slice is SHA256 hashed, then the result is RIPEMD160 hashed.
+func Sha256RipeMD160(b []byte) []byte {
+	sha256 := sha256.New()
+	sha256.Write(b)
+
+	// TODO: Check if this is still needed
```
Comment (AdriaCarrera) [inline address-codec/sha256.go:16]: Can we delete this TODO?

File: address-codec/sha256.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	//nolint:gosec // G406: Use of deprecated weak cryptographic primitive (gosec)
```
Comment (AdriaCarrera) [inline address-codec/sha256.go:17]: If deprecated, is there a newer version of ripemd?

File: coverage.html [hunk 4]
```diff
@@ -55,391 +55,393 @@
 			<div id="nav">
 				<select id="files">
 				
-				<option value="file0">github.com/Peersyst/xrpl-go/address-codec/address_codec.go (93.9%)</option>
+				<option value="file0">github.com/Peersyst/xrpl-go/address-codec/base58.go (100.0%)</option>
```
Comment (AdriaCarrera) [inline coverage.html:58]: Should this file be committed to the repo?

**Review by AdriaCarrera — APPROVED** [review]
LGTM

**Pull request merged**